### PR TITLE
feat(pie-list): DSW-4117 link list-items

### PR DESCRIPTION
**Pull request by jamieomaguire**
## Describe your changes (can list changeset entries if preferable)

- Enables list-items to serve as links. Simply provide the `isLink` property and an `<a>` tag (without text inside) to a named `link` slot. The list item will construct appropriate aria using it's text properties so that the link is appropriately narrated. This is overridable by consumers providing their own aria. 
- This approach also allows consumers to use the Link components provided by larger frameworks such as Nuxt and NextJS. This is because those components render as `a` tags in the browser.

https://github.com/justeattakeaway/pie-aperture/pull/546

## Author Checklist (complete before requesting a review, do not delete any)
- [x] I have performed a self-review of my code.
- [x] I have added thorough tests where applicable (unit / component / visual).
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview.
- [x] I have reviewed visual test updates properly before approving.
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have tested these changes in [PIE Aperture](https://github.com/justeattakeaway/pie-aperture/) using the `/test-aperture` command.

## Not-applicable Checklist items
Please move any Author checklist items that do not apply to this pull request here.

---

## Reviewer checklists (complete before approving)
Mark items as `[-] N/A` if not applicable.

### Reviewer 1 - @fernandofranca 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview.
- [x] I have verified that all acceptance criteria for this ticket have been completed.
- [x] I have reviewed the Aperture changes (if added)
- [x] If there are visual test updates, I have reviewed them.

### Reviewer 2
- [-] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview.
- [-] I have verified that all acceptance criteria for this ticket have been completed.
- [-] I have reviewed the Aperture changes (if added)
- [-] If there are visual test updates, I have reviewed them.

### CODE REVIEW

**Review by fernandofranca — DISMISSED** [review]

File: packages/components/pie-list/src/pie-list-item/index.ts [hunk 1]
```diff
@@ -155,6 +180,48 @@ export class PieListItem extends PieElement implements ListItemProps {
...
+     * owns the child's aria. The anchor is plain light-DOM, so it cannot reference the shadow-DOM text
+     * via `aria-labelledby`; and it is not a PIE control that consumes `ariaContext`, so we set the
+     * attributes imperatively. The visible text is `aria-hidden` in the template, so nothing is
+     * announced twice.
+     *
+     * A consumer-provided name/description always wins: we capture (once) whether the anchor already
+     * carries `aria-label`/`aria-labelledby` or `aria-description`/`aria-describedby` and, if so,
+     * leave that attribute untouched. Otherwise we keep our generated value in sync with the text.
+     */
+    private _applyLinkAria (): void {
```
Comment (fernandofranca) [inline packages/components/pie-list/src/pie-list-item/index.ts:199]: Since this an imperative approach to set and remove attributes, it might be the case when `_isLinkRow` was true, but then for some reason flipped to false, the attributes won't be removed.

It's kind of an edge case, though.

File: apps/pie-storybook/stories/testing/pie-list.test.stories.ts [hunk 2]
```diff
@@ -98,23 +98,48 @@ const SwitchSelectionTemplate = () => {
...
+// stretched over the row. The four items cover the text combinations (both, secondary only, meta
+// only, neither) so we can assert the anchor's derived aria-label/aria-description, that the visible
+// text is aria-hidden, and that clicking anywhere on the row activates the link.
+const LinkListTemplate = () => withLayout(html`
+    <pie-list aria-label="Manage your restaurant">
+        <pie-list-item .interactionType=${'link'} data-test-id="item-1" primaryText="Orders" secondaryText="View and manage live orders" metaText="12 active">
+            <a slot="link" href="#orders" data-test-id="link-1"></a>
+        </pie-list-item>
+        <pie-list-item .interactionType=${'link'} data-test-id="item-2" primaryText="Menu" secondaryText="Edit items and prices">
+            <a slot="link" href="#menu" data-test-id="link-2"></a>
```
Comment (fernandofranca) [inline apps/pie-storybook/stories/testing/pie-list.test.stories.ts:128]: Should there be a way of representing and handling disabled link list items?

Comment (jamieomaguire) [reply apps/pie-storybook/stories/testing/pie-list.test.stories.ts:128]: @fernandofranca good question! For now I think we don't need to worry about this - the designs do not cater to it either. However we can fix forward if a need is raised by consumers.

**Review by fernandofranca — APPROVED** [review]
